rpkg_src/decrypt_packagedefinition_thumbs.cpp

```diff
@@ -21,16 +22,21 @@ void generic_function::decrypt_packagedefinition_thumbs(std::string& input_path,
 
     uint64_t packagedefinitions_thumbs_file_size = file.tellg();
 
-    int packagedefinitions_thumbs_header_size = 20;
+    int packagedefinitions_thumbs_header_size = 16;
 
-    packagedefinitions_thumbs_file_size -= packagedefinitions_thumbs_header_size;
+	packagedefinitions_thumbs_file_size -= packagedefinitions_thumbs_header_size;
+	packagedefinitions_thumbs_file_size -= 4; // -4 to skip the checksum
 
     file.seekg(0, file.beg);
 
     std::vector<char> packagedefinitions_thumbs_header(packagedefinitions_thumbs_header_size, 0);
 
     file.read(packagedefinitions_thumbs_header.data(), packagedefinitions_thumbs_header_size);
 
+	std::vector<char> checksum_raw(4, 0);
+
+	file.read(checksum_raw.data(), 4);
+
```

Comment:
```suggestion
    std::vector<char> checksum_raw(4, 0);

    file.read(checksum_raw.data(), 4);

```

Reply:
Blame Visual Studio. Fixed in another commit.